src/main/java/it/pagopa/ecommerce/commons/utils/RedirectConfigurationKeysConfig.java

```diff
@@ -0,0 +1,39 @@
+package it.pagopa.ecommerce.commons.utils;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class RedirectConfigurationKeysConfig {
+
+    private final Map<String, URI> redirectBeApiCallUriMap;
+
+    public RedirectConfigurationKeysConfig(Map<String, URI> redirectBeApiCallUriMap) {
+        this.redirectBeApiCallUriMap = redirectBeApiCallUriMap;
+    }
+
+    /**
+     * Execute a recursive search on the redirectBeApiCallUriMap. The recursion
+     * method will be called with the key without the first parameter element.
+     *
+     * @param params List of parameters that compose the key.
+     * @return The found URI or an empty value.
+     */
+    public Optional<URI> searchRedirectUrlForPsp(String... params) {
```

Comment:
make this method private and add a public method with search parameter that returns an `Either<RedirectConfigurationException, URI>` as done [here](https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/6fd2d721add5a0b66a1bf5550d605b2f8b2eff17/src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java#L700).

Add also exception class so that all that logic present into transactions service can be moved out here and re-used by both transactions-service and event-dispatcher

Reply:
Done.